karma.conf.js

```diff
@@ -2,17 +2,42 @@ module.exports = (config) => {
   config.set({
     frameworks: ['jasmine'],
     files: [
+      'src/index.js',
       'src/**/*_test.js',
     ],
     preprocessors: {
-      'src/**/*_test.js': ['webpack'],
+      'src/index.js': ['webpack'],
```

Comment:
We had this config wrong the whole time? Why aren't the `*_test.js` files being processed?

Reply:
So what was happening is each `*_test.js` file was including all of the dependencies and webpacking that. That would probably be fine, but for some reason tfjs was struggling with that - somehow the Tensors from one instance were going through functions from the other instance and not being checked correctly, as they wouldn't be an instance of the Tensor class from that version of tfjs. This just builds the dependencies of all test suites (aka the whole library) and includes that first, so you only get one version of all the deps

Reply:
When you pass a Tensor through multiple instances of TensorFlow.js, we fail when doing input validation for op calls because "instanceof Tensor" refers to the wrong Tensor instance (from the other instantiation of TensorFlow.js). This problem is artificial and not actually fundamental -- I will send an PR to fix this in TensorFlow.js core, however it's less pressing now that you've solved it here.

Reply:
amazing, thanks for the detailed description @meiamsome!